File: _old/2024/2 - person filter/personFilter_start.js
// generate a structured report that lists all adult individuals (age > 18),
// grouped by their city of residence,
// and ordered alphabetically by name within each city.

const assert = require("assert");

class Person {
  constructor(name, age, city) {
    this.name = name;
    this.age = age;
    this.city = city;
  }
}

// Function to be implemented
function filterGroupAndSortPersons(persons) {
  return persons;
}

// Example Usage
const persons = [
  new Person("John", 25, "New York"),
  new Person("Alice", 17, "New York"),
  new Person("Bob", 20, "Los Angeles"),
  new Person("Emma", 19, "New York"),
  new Person("Chris", 15, "Chicago"),
  new Person("Anna", 30, "New York"),
  new Person("Zoe", 22, "New York"),
];

const groupedAndSortedPersons = filterGroupAndSortPersons(persons);
const expectedResponse = {
  "New York": [
    { name: "Anna", age: 30 },
    { name: "Emma", age: 19 },
    { name: "John", age: 25 },
    { name: "Zoe", age: 22 },
  ],
  "Los Angeles": [{ name: "Bob", age: 20 }],
};
console.log(groupedAndSortedPersons);
console.log(expectedResponse);
//assert.deepEqual(groupedAndSortedPersons, expectedResponse);

File: _old/2024/2 - person filter/personFilter_start.c
// generate a structured report that lists all adult individuals (age > 18),
// grouped by their city of residence,
// and ordered alphabetically by name within each city.

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_NAME_LENGTH 50
#define MAX_CITY_LENGTH 50

typedef struct {
    char name[MAX_NAME_LENGTH];
    int age;
    char city[MAX_CITY_LENGTH];
} Person;

// Function to be implemented
void filter_group_and_sort_persons(Person* persons, int person_count) {

}

int main() {
    Person persons[] = {
        {"John", 25, "New York"},
        {"Alice", 17, "New York"},
        {"Bob", 20, "Los Angeles"},
        {"Emma", 19, "New York"},
        {"Chris", 15, "Chicago"},
        {"Anna", 30, "New York"},
        {"Zoe", 22, "New York"}
    };
    int person_count = sizeof(persons) / sizeof(persons[0]);

    filter_group_and_sort_persons(persons, person_count);


    return 0;
}
